### PR TITLE
AdditionalInfo changes for Baseline Control

### PR DESCRIPTION
**Pull request by v-dpardeshi**
## Description
</br>
 Brief description of the requested change.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1 [hunk 1]
```diff
@@ -127,6 +127,7 @@ class AgentPool: ADOSVTBase
             {
                 if ($this.AgentPoolOrgObj.autoProvision -eq $true) {
                     $controlResult.AddMessage([VerificationResult]::Failed,"Auto-provisioning is enabled for the $($this.AgentPoolOrgObj.name) agent pool.");
+                    $controlResult.AdditionalInfoInCSV += "NA";
```
Comment (juhi037) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1:130]: Add NA for passing scenarios as well (just AdditionalInfoInCSV). Also do we have 'AdditionalInfo' in this control similar to the one you added in auto update control? If not we should add it here as well.

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1 [hunk 2]
```diff
@@ -212,6 +213,8 @@ class AgentPool: ADOSVTBase
                 else
                 {
                     $controlResult.AddMessage([VerificationResult]::Failed,"Auto-update of agents is disabled for [$($this.AgentPoolOrgObj.name)] agent pool.");
+                    $controlResult.AdditionalInfo = "Auto-update of agents is disabled for [$($this.AgentPoolOrgObj.name)] agent pool.";
+                    $controlResult.AdditionalInfoInCSV = "NA";
                 }
```
Comment (juhi037) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1:218]: Similar as above (just additional info in csv not AdditionalInfo)

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.CommonSVTControls.ps1 [hunk 3]
```diff
@@ -683,6 +685,7 @@ class CommonSVTControls: ADOSVTBase {
                 if ($this.ControlFixBackupRequired){
                     $controlResult.BackupControlState = $secureFilePipelinePermObj;
                 }
+                $controlResult.AdditionalInfoInCSV = "Secure file is accesible to all YAML pipelines";
```
Comment (juhi037) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.CommonSVTControls.ps1:688]: AdditionalInfoInCSV should be NA in this case. Similar in passing scenario.

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.CommonSVTControls.ps1 [hunk 4]
```diff
@@ -558,6 +558,8 @@ class CommonSVTControls: ADOSVTBase {
                     $controlResult.SetStateData("List of groups: ", $feedWithBroaderGroup);
                     $groups = $feedWithBroaderGroup | ForEach-Object { $_.Name + ': ' + $_.Role } 
                     $controlResult.AdditionalInfoInCSV = $groups -join ' ; '
+                    $controlResult.AdditionalInfo += "Count of broader groups that have administrator/contributor access to feed: $($feedWithBroaderGroupCount)";
+                    $controlResult.AdditionalInfo += "List of Broader groups" + $groups ;
```
Comment (juhi037) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.CommonSVTControls.ps1:562]: List of broader groups:

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.CommonSVTControls.ps1 [hunk 5]
```diff
@@ -790,6 +793,7 @@ class CommonSVTControls: ADOSVTBase {
                     }
 
                     $groups = $secureFileWithBroaderGroup | ForEach-Object { $_.Name + ': ' + $_.Role } 
+                    $controlResult.AdditionalInfo = "Count of broader groups that have user/administrator access to secure file: $($secureFileWithBroaderGroupCount)"
```
Comment (juhi037) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.CommonSVTControls.ps1:796]: not consistent with the changes in feeds broader grp control. (list of broader grps not added in additionalinfo here, but is it added in the former control)

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.ServiceConnection.ps1 [hunk 6]
```diff
@@ -264,6 +264,7 @@ class ServiceConnection: ADOSVTBase
                     $controlResult.AddMessage("Global groups that have access to service connection.",$restrictedGroups)
                     $controlResult.SetStateData("Global groups that have access to service connection",$restrictedGroups)
                     $controlResult.AdditionalInfo += "Count of global groups that have access to service connection: " + @($restrictedGroups).Count;
+                    $controlResult.AdditionalInfoInCSV ="global groups that have access to service connection."
```
Comment (juhi037) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.ServiceConnection.ps1:267]: Global groups.... Also the list of global groups is not mentioned. Seems incomplete?

**Review by juhi037 — APPROVED** [review]

**Review by v-soukum — APPROVED** [review]
looks good

**Pull request merged**